src/node_snapshotable.cc

```diff
@@ -349,15 +354,15 @@ class FileWriter : public FileIO {
     size_t written_total = Write<size_t>(data.size());
     if (is_debug) {
       std::string str = ToStr(data);
-      Debug("WriteString(), length=%d: \"%s\"\n", data.size(), data.c_str());
+      Debug("WriteString(), length=%zu: \"%s\"\n", data.size(), data.c_str());
     }
 
-    size_t r = fwrite(data.c_str(), 1, data.size() + 1, f);
-    CHECK_EQ(r, data.size() + 1);
-    written_total += r;
+    // Write the null-terminated string.
+    sink.insert(sink.end(), data.c_str(), data.c_str() + data.size() + 1);
+    written_total += data.size();
```

Comment:
I realize this is pre-existing but... why not `written_total += data.size() + 1;`?

Reply:
The layout of a written string already contains the length of the data. It seems like the null terminator is not necessary here.

Reply:
The original `r` should be `data.size() + 1`, so this should be updated now, thanks for catching it :)

I still prefer to keep the null terminator as sometimes that helps debugging the de/serialization (when something goes wrong...).

Reply:
Updated to `data.size() + 1`